lib/dpl/provider/bintray.rb

```diff
@@ -0,0 +1,532 @@
+require 'json'
+require 'net/http'
+require "uri"
+require 'find'
+
+module DPL
+  class Provider
+    class Bintray < Provider
+
+      @file
+      @user
+      @key
+      @passphrase
+      @url
+      @descriptor
+      @dry_run
+      @test_mode = false
+
+      def check_auth
+      end
+
+      def needs_key?
+        false
+      end
+
+      def set_test_mode
+        @test_mode = true
+      end
+
+      def init_from_args
+        @user = options[:user]
+        @key = options[:key]
+        @url = options[:url]
+        @file = options[:file]
+        @passphrase = options[:passphrase]
+        @dry_run = options[:dry_run]
+
+        if @user.nil?
```

Comment:
I *think* `@user` can get the value `false` in this context, as unlikely as it might be.